marvel_superheroes/src/components/ComicsPage/ComicsPage.tsx

```diff
@@ -62,32 +58,25 @@ class ComicsPage extends React.Component<IProps, IState> {
         this.props.history.push('?page=' + page);
     }
 
-    setCurrentPage = (currentPage: number): void => {
-        this.setState({
-            currentPage: currentPage,
-            offset: ((currentPage - 1) * 4)
-        });
-        this.addressBarMaker(currentPage);
+    setCurrentPage = (currentComicsPage: number): void => {
+        this.props.setComicsCurrentPage(currentComicsPage);
+        this.addressBarMaker(currentComicsPage);
     }
 
     makeRequest = async (): Promise<void> => {
-        this.setState({ progresBar: true });
-        const ComicsHeroResult = await getComicsHero(this.props);
-        const comicsResult = await getComics(this.state, this.props);
-        this.setState({
-            heroName: ComicsHeroResult.data.data.results[0].name,
-            comics: comicsResult.data.data.results,
-            totalOfItems: comicsResult.data.data.total,
-            progresBar: false
-        });
+        const page = queryString.parse(this.props.location.search).page;
+        const heroId = this.props.match.params.id;
+        const currentComicsPage = this.props.currentComicsPage;
```

Comment:
you can use destructuring to get props value and not to write this.props... all the time